### PR TITLE
test(cache): verify TTLCache behavior for empty string keys (Variation 2)  (#1402)

### LINKED ISSUE #1402: test(cache): verify TTLCache behavior for empty string keys (Variation 2)
## 🧩 Background
The existing cache implementation must handle boundary inputs robustly. Currently, tests do not comprehensively cover extreme values.

## 🎯 Objective
Add a specific test to verify cache behavior for empty string keys.

## 📁 Files to touch
- `lib/cache.test.ts`

## 🛠️ Implementation steps
- Assert that setting a value with '' rejects with an empty key error.
- Verify verify that an empty cache key is not allowed.
- Run Vitest to ensure the test coverage is successfully verified.

## ✅ Definition of done
- [ ] 1 new test case targeting this boundary in cache.
- [ ] All cache test suites pass successfully.

### PR DESCRIPTION
**Pull request by Master-Bruce-Wayne**
## Description

Fixes #1402 

## Pillar

- [ ] 🎨 Pillar 1 — New Theme Design
- [ ] 📐 Pillar 2 — Geometric SVG Improvement
- [ ] 🕐 Pillar 3 — Timezone Logic Optimization
- [x] 🛠️ Other (Bug fix, refactoring, docs)

## Checklist before requesting a review:

- [x] I have read the `CONTRIBUTING.md` file.
- [x] I have tested these changes locally (`localhost:3000/api/streak?user=YOUR_USERNAME`).
- [x] I have run `npm run format` and `npm run lint` locally and resolved all errors (CI will fail otherwise).
- [x] My commits follow the Conventional Commits format (e.g., `feat(themes): ...`, `fix(calculate): ...`).
- [ ] I have updated `README.md` if I added a new theme or URL parameter.
- [x] I have started the repo.
- [x] I have made sure that i have only one commit to merge in this PR.
- [x] The SVG output matches the CommitPulse "premium quality" aesthetic standard (no raw elements, smooth animations, correct fonts).
- [x] (Recommended) I joined the CommitPulse Discord community for contributor discussions, mentorship, and faster PR support.

---

I have tested every tests for these changes. 

Screenshot 2026-06-01 020511

### CODE REVIEW

File: lib/cache.test.ts [hunk 1]
```diff
@@ -500,6 +500,34 @@ describe('TTLCache', () => {
       cache.destroy();
     });
 
+    it('verify TTLCache behavior for empty string keys (Variation 2)', () => {
```
[the remark this answers is not in the document]
Comment (Master-Bruce-Wayne) [reply lib/cache.test.ts:507]: let me change that
